lib/mayaUsd/ufe/Global.cpp

```diff
@@ -193,6 +196,9 @@ MStatus initialize()
     g_USDRtid = Ufe::RunTimeMgr::instance().register_(
         kUSDRunTimeName, usdHierHandler, usdTrans3dHandler, usdSceneItemOpsHandler);
 #endif
+#ifdef UFE_V4_FEATURES_AVAILABLE
+    handlers.nodeDefHandler = UsdShaderNodeDefHandler::create();
```

Comment:
This can't work, you're adding the node definition handler to the handlers struct after registration is done at line 183.